File: LBOCExpandFriendList/LBExpandFriendList/home/LBHomeTableView.m
//
//  LBHomeTableView.m
//  LBExpandFriendList
//
//  Created by yunmei on 2018/1/20.
//  Copyright © 2018年 yunmei. All rights reserved.
//

#import "LBHomeTableView.h"
#import "LBHomeCell.h"
#import "UIView+ViewController.h"
#import "LBExpandController.h"
#import "LBQQFriendListController.h"
@implementation LBHomeTableView

-(instancetype)initWithFrame:(CGRect)frame style:(UITableViewStyle)style{
    
    self = [super initWithFrame:frame style:style];
    if (self) {
        self.delegate = self;
        self.dataSource = self;
    }
    return self;
}
-(void)setDataSourceArray:(NSArray *)dataSourceArray{
    _dataSourceArray = dataSourceArray;
    [self reloadData];
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return _dataSourceArray.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    LBHomeCell *cell = [LBHomeCell cellWithTableView:tableView];
    cell.titleLabel.text = [NSString stringWithFormat:@"%@",_dataSourceArray[indexPath.row]];
    return cell;
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if(indexPath.row == 0){
        [self.viewController.navigationController pushViewController:[[LBExpandController alloc]init] animated:YES];
    }else{
        [self.viewController.navigationController pushViewController:[[LBQQFriendListController alloc]init] animated:YES];
    }
    
    
}
@end

File: LBSwiftExpandFriendList/LBSwiftExpandFriendList/home/LBHomeTableView.swift
//
//  LBHomeTableView.swift
//  LBSwiftExpandFriendList
//
//  Created by yunmei on 2018/1/20.
//  Copyright © 2018年 liubo. All rights reserved.
//

import UIKit

class LBHomeTableView: UITableView {

    lazy var dataSourceArray:[String] = [String]()
    override init(frame: CGRect, style: UITableViewStyle) {
        super.init(frame: frame, style: style)
        self.delegate = self
        self.dataSource = self
        self.register(LBHomeCell.self, forCellReuseIdentifier: "LBHomeCell")
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
extension LBHomeTableView:UITableViewDelegate,UITableViewDataSource{
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSourceArray.count
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "LBHomeCell") as! LBHomeCell
        cell.titleLabel.text = "\(dataSourceArray[indexPath.row])"
        return cell;
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        if indexPath.row == 0 {
            viewController().navigationController?.pushViewController(LBExpandController(), animated: true)
        }else{
            viewController().navigationController?.pushViewController(LBQQFriendListController(), animated: true)
        }
    }

}
